File: src/modules/belt/domain/belt-progress.entity.ts
import { User } from "src/modules/user/domain/user.entity";
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class BeltProgress {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => User, (user) => user.beltProgress, { onDelete: 'CASCADE' })
  user: User;

  @Column({ type: 'enum', enum: ['WHITE', 'BLUE', 'PURPLE', 'BROWN', 'BLACK'] })
  belt: string;

  @Column({ type: 'int', default: 0 })
  stripes: number;

  @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  achievedAt: Date;
}

File: src/modules/belt/domain/belt_progress_entity.go
package domain_belt

import "time"

type Belt int

const (
	White Belt = iota
	Blue
	Purple
	Brown
	Black
)

type BeltProgress struct {
	ID          int       `json:"id" gorm:"primaryKey, autoIncrement"`
	UserID      string    `json:"user_id" gorm:"not null"`
	User        string    `json:"user" gorm:"not null"`
	CurrentBelt Belt      `json:"current_belt" gorm:"not null"`
	StripeCount int       `json:"stripe_count" gorm:"default:0"`
	EarnedAt    time.Time `json:"earned_at" gorm:"default:CURRENT_TIMESTAMP"`
}
